setup.go

```diff
@@ -45,62 +64,63 @@ var loggerMutex sync.RWMutex
 var loggers = make(map[string]*zap.SugaredLogger)
 var levels = make(map[string]zap.AtomicLevel)
 
+// loggerCore is the base for all loggers created by this package
+var loggerCore = &lockedMultiCore{}
+
+// primaryFormat is the default format of the primary core used for logging
+var primaryFormat LogFormat = ColorizedOutput
+
+// primaryLevel is the default log level of the primary core used for logging
+var primaryLevel LogLevel = LevelError
+
+var primaryCoreMu sync.Mutex // guards changes to the primaryCore variable
+
+// primaryCore is the primary logging core
+var primaryCore zapcore.Core
+
 // SetupLogging will initialize the logger backend and set the flags.
 // TODO calling this in `init` pushes all configuration to env variables
 // - move it out of `init`? then we need to change all the code (js-ipfs, go-ipfs) to call this explicitly
 // - have it look for a config file? need to define what that is
-var zapCfg = zap.NewProductionConfig()
+func SetupLogging(cfg Config) {
+	primaryFormat = cfg.Format
+	primaryLevel = cfg.Level
```

Comment:
* We need to either protect `primaryLevel` with `loggerMutex`, or use an atomic level.
* We should also protect `primaryFormat` with some form of lock.

The simplest solution is probably to just:

1. Get rid of `primaryCoreMu`.
2. Protect _all_ this state with `loggerMutex`.

We could use finer-grained locks, but this is setup code so it doesn't really matter.

Reply:
Good catch. Done